pkg/infra/remotecache/remotecache.go

```diff
@@ -131,17 +138,46 @@ type cachedItem struct {
 	Val interface{}
 }
 
-func encodeGob(item *cachedItem) ([]byte, error) {
+type codec interface {
```

Comment:
nit: Can you put this `codec` interface and implementations into another file? `codec.go`